runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsPlugin.java

```diff
@@ -175,4 +183,36 @@ else if (config.drawNonClanMemberNames() && !player.isClanMember())
 			}
 		}
 	}
+
+	public boolean isWithinLevelRange(int playerCombatLevel)
+	{
+		Widget levelRangeWidget = client.getWidget(WidgetInfo.COMBAT_AREA_LEVEL_RANGE);
+		Widget wildernessLevelWidget = client.getWidget(WidgetInfo.COMBAT_AREA_WILDERNESS_LEVEL);
+
+		int localPlayerLevel = client.getLocalPlayer().getCombatLevel();
+		int lowerLevelBound = localPlayerLevel - 15;
+		int upperLevelBound = localPlayerLevel + 15;
+
+		if (levelRangeWidget == null && wildernessLevelWidget == null)
+		{
+			return false;
+		}
+		else if (!levelRangeWidget.isHidden() && !wildernessLevelWidget.isHidden())
+		{
+			lowerLevelBound = Integer.parseInt(levelRangeWidget.getText().split("-")[0]);
```

Comment:
The wilderness level is likely in a varbit... it would be better to read it from there than the the widget.

Reply:
I had a look into varbits using 'Get Settings' at level 1 wilderness and running to level 2 wilderness and clicking 'Get Settings' again. I couldn't find a change but the thing that did change was whether the user entered the wilderness and left it or vice versa. It didn't take into account at what level they were at.

Reply:
This is unfortunate. It looks like the client computes the wilderness level itself. @Joshua-F provided some decompiled cs2 that the client uses (we do not have a decompiler)

```
[proc,wilderness_level](int)
$int0 = 0;
if(~inzone(48238016, 860868863, coord) == 1) {
    $int0 = add(divide(sub(coordz(coord), multiply(55, 64)), 8), 1);
} else if(~inzone(49293184, 855631807, coord) == 1) {
    $int0 = sub(divide(sub(coordz(coord), multiply(155, 64)), 8), 1);
} else if(~inzone(48244416, 860875391, coord) == 1) {
    $int0 = add(divide(sub(coordz(coord), multiply(155, 64)), 8), 1);
}
return($int0);
```

```
[proc,inzone](int $int0, int $int1, int $int2)(int)
$int3 = coordx($int2);
if($int3 < coordx($int0) || $int3 > coordx($int1)) {
    return(0);
}
$int3 = coordy($int2);
if($int3 < coordy($int0) || $int3 > coordy($int1)) {
    return(0);
}
$int3 = coordz($int2);
if($int3 < coordz($int0) || $int3 > coordz($int1)) {
    return(0);
}
return(1);
```

where `coord(x,y,z) = y | x << 14 | z << 28`.

I would prefer writing this logic in Java and using that.